src/templates/Pool/NewPoolOperations/Form/Invest/index.tsx

```diff
@@ -120,6 +122,14 @@ const Invest = ({ typeAction, privateInvestors }: IInvestProps) => {
   const { txNotification, transactionErrors } = useTransaction()
   const router = useRouter()
   const { data: pool } = usePoolData({ id: router?.query.address as string })
+  const { data } = usePoolInfo({
+    id: pool?.id || '',
+    day: Math.trunc(Date.now() / 1000 - 60 * 60 * 24)
+  })
+  const { data: referralData } = useReferralDecrypt({
+    hash: router.query.referral as string | undefined,
```

Comment:
This `router.query.referral` can be an array, you should not ignore it in my opinion, it's verbose but you can do it like this, and be sure its an `array` or a `string`

```suggestion
    hash: typeof router.query.referral === "string" ? router.query.referral : undefined,
```